mlir/tools/mlir-miopen-lib/CMakeLists.txt

```diff
@@ -130,10 +124,15 @@ if( BUILD_FAT_LIBMLIRMIOPEN )
 
     # Install lib${target_name}.a to ${CMAKE_INSTALL_PREFIX}/lib/<package>/
     # and group it into the ${export-set}
+    # The INTERFACE_INCLUDE_DIRECTORIES will be set as the property of the exported library
+    # so that when comsumer libraries (MIOpen) link in ibMLIRMIOpen, ${INCLUDE_DIR} is
+    # automatically propagated to the consumer library's include_dirs, which has the same
+    # effects as calling target_include_directories(MIOpen ${MLIRMIOpen_INCLUDE_DIR})
     install(TARGETS ${target_name}
       EXPORT ${export_set}
       ARCHIVE DESTINATION lib/${PACKAGE_NAME}
-      COMPONENT ${component_name})
+      COMPONENT ${component_name}
+      INCLUDES DESTINATION ${INCLUDE_DIR})
```

Comment:
Just for clarification, is it the `INCLUDES DESTINATION` that will set `INTERFACE_INCLUDE_DIRECTORIES`? Also, which exported cmake file will help MIOpen to automatically find the `${INCLUDE_DIR}`?

Reply:
Yes.
The `find_package()` in MIOpen first locates `MiirConfig.cmake`, which includes `MiirTargets.cmake`, which further set the `INTERFACE_INCLUDE_DIRECTORIES` property of the `Miir::MLIRMIOpen` targets as follows
```cmake
add_library(Miir::MLIRMIOpen STATIC IMPORTED)
set_target_properties(Miir::MLIRMIOpen PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include/Miir")
```
Later in MIOpen, when `target_link_libraries(MIOpen Miir::MLIRMIOpen)` is called, MIOpen obtains MLIRMIOpen's `INTERFACE_INCLUDE_DIRECTORIES` automatically.

Reply:
Make sense and thanks for the detailed explanation!